docs/key_format.md

```diff
@@ -0,0 +1,36 @@
+# Key Format
+
+Nodes, orphans, and roots are stored under the database with different key formats to ensure there are no key collisions and a structured key from which we can extract useful information.
+
+### Nodes
+
+Node KeyFormat: `n|<node.hash>`
+
+Nodes are marshalled and stored under nodekey with prefix `n` to prevent collisions and then appended with the node's hash.
+
+### Orphans
+
+Orphan KeyFormat: `o|toVersion|fromVersion|hash`
+
+Orphans are marshalled nodes stored with prefix `o` to prevent collisions. You can extract the toVersion, fromVersion and hash from the orphan key by using:
+
+```golang
+// orphanKey: o|50|30|0xABCD
+var toVersion, fromVersion int64
+var hash []byte
+orphanKeyFormat.Scan(orphanKey, &toVersion, &fromVersion, hash)
```

Comment:
What does this `orphanKeyFormat.Scan(orphanKey, &toVersion, &fromVersion, hash)` function do? Just break up the key on `|`?

Reply:
Yea the orphan Key has an expected format `o|toVersion|fromVersion|hash`. `Scan` will use this expected format to parse the key and place the toVersion, fromVersion, hash encoded in the key into the respective pointers that are passed in